test/contrib/test_socks.py

```diff
@@ -229,6 +229,37 @@ def request_handler(listener):
         self.assertEqual(response.data, b'')
         self.assertEqual(response.headers['Server'], 'SocksTestServer')
 
+    def test_local_dns(self):
+        def request_handler(listener):
+            sock = listener.accept()[0]
+
+            handler = handle_socks5_negotiation(sock, negotiate=False)
+            addr, port = next(handler)
+
+            self.assertIn(addr, ['127.0.0.1', '::1'])
```

Comment:
We don't just run in Python 2.6. We run in Python 2.6, Python 2.7, Python 3.3, Python 3.4, Python 3.5, and PyPy 2, across two different operating systems. =) The tests, and the code, must work on all of those platforms.